TESTING.md

```diff
@@ -46,19 +46,28 @@ plugin, create a mailing with a block of content. Then try each of the following
  3. Edit a paragraph. Press Ctrl-Shift-T and see a dialog. Enter a filter and pick a token using the keyboard. Observe the new token in the paragraph.
  4. Edit a heading or button. Ensure that the the token icon/dropdown/hotkey work as expected. Observe the new token in the heading or button.
 
-## Manual Tests: Images
+## Manual Tests: Images and Links
 
-Mosaico handles a few different kinds of images. To test these, create a
+Mosaico handles a few different kinds of images and links. To test these, create a
 mailing then:
 
- 1. Add blocks with images
+ 1. Add blocks
    1. Add a block which supports an image. Upload an image.
-   2. Add a block which supports an image. Re-use the previously uplaoded image.
-   3. Add a footer block which uses the built-in Twitter/Facebook icons
- 2. Test the mailing. Check:
-   1. The first block should show the uploaded image. The URL should be absolute.
-   2. The second block should show the same uploaded image. The URL should be absolute.
-   3. The footer block should show the Twitter/Facebook icons. The URL should be absolute.
+   2. Add a block which supports an image. Re-use the previously uploaded image.
+   3. Add a footer block which uses the built-in Twitter/Facebook icons.
+   4. In one of the blocks, add a hyperlink to an external page (eg `https://google.com`).
+   5. In one of the blocks, add a hyperlink to an internal page (eg `http://dmaster.l/user`).
+   6. In one of the blocks, add a hyperlink to using a token (eg `{action.forward}`)
+ 2. (Definition: The "Message Evaluation Procedure" is as follows. We will reference this in subsequent steps.)
+   1. Check the the uploaded image. The image should appear and use an absolute URL.
+   2. check the the re-used uploaded image. The image should appear and use an absolute URL.
+   3. Check the the Twitter/Facebook icons. The image should appear and use an absolute URL.
+   4. Check the hyperlink to the external page (eg `https://google.com`). The link should be absolute. (For finalized mailing, the link should work.)
+   5. Check the hyperlink to the internal page (eg `http://dmaster.l/user`). The link should be absolute. (For finalized mailing, the link should work.)
+   6. Check the hyperlink created via token. The link should be absolute. (For finalized mailing, the link should work.)
+ 3. Using the "Test", open the "Preview" in HTML. Perform the "Message Evaluation Procedure".
+ 4. Using the "Test", send a message to an email address. Perform the "Message Evaluation Procedure".
+ 5. Finalize and submit the mailing. Trigger cron (eg `cv api -U admin job.process_mailing`). Perform the "Message Evaluation Procedure".
```

Comment:
Need another step here:
1. From your mail client, click on the "View in Browser" link. Perform the "Message Evaluation Procedure".

Reply:
It's no wonder we're playing a game of whack-a-mole on this -- there are *four* ways to render a message. Thanks for pointing this one out.